src/cli/commands/force/lightning/local/device/create.ts

```diff
@@ -30,31 +33,46 @@ const messages = Messages.loadMessages(
     'device-create'
 );
 
-export class Create extends Setup {
+export class Create extends SfdxCommand implements HasRequirements {
     public static description = messages.getMessage('commandDescription');
 
     public static readonly flagsConfig: FlagsConfig = {
-        apilevel: flags.string({
-            char: 'a',
-            description: messages.getMessage('apiLevelFlagDescription'),
-            longDescription: messages.getMessage('apiLevelFlagDescription'),
-            required: false
-        }),
         devicename: flags.string({
             char: 'n',
             description: messages.getMessage('deviceNameFlagDescription'),
-            required: true
+            required: true,
+            validate: (deviceName) => {
+                if (deviceName && deviceName.trim().length > 0) {
+                    return true;
+                } else {
+                    throw new SfdxError(
+                        messages.getMessage(
+                            'error:invalidDeviceNameFlagsDescription'
+                        ),
+                        'lwc-dev-mobile'
+                    );
+                }
+            }
         }),
         devicetype: flags.string({
             char: 'd',
             description: messages.getMessage('deviceTypeFlagDescription'),
-            required: true
+            required: true,
+            validate: (deviceType) => {
+                if (deviceType && deviceType.trim().length > 0) {
+                    return true;
+                } else {
+                    throw new SfdxError(
+                        messages.getMessage(
+                            'error:invalidDeviceTypeFlagsDescription'
+                        ),
+                        'lwc-dev-mobile'
+                    );
+                }
+            }
         }),
-        platform: flags.string({
-            char: 'p',
-            description: messages.getMessage('platformFlagDescription'),
-            required: true
-        })
+        ...CommandLineUtils.createFlagConfig(FlagsConfigType.ApiLevel, false),
```

Comment:
What does the ellipsis operator do in this case? I'm used to that as a rest or spread operator, but it's not clear how either applies in this usage.

Reply:
Oh, does it denormalize the key/value aspects of a `FlagsConfig` object, out to this enjoining object?

Reply:
Yes, it's a rest destructuring for object. `createFlagConfig` return an object with one key, either `platform` or `apilevel`, with respective properties.